### PR TITLE
fix(ts-typings): make define generic so it returns proper constructor

### LINKED ISSUE #1050: ts-definitions: make define generic
Currently `define` accepts just `Function` type and returns `any` which is kinda cumbersome, when used with anonymous classes, because we loose proper type from it.

now:
```ts
 const Ctor2 = skate.define(class extends HTMLElement {
    static is = 'x-test-2';
    foo: number;
  });

const inst = new Ctor2();
console.log(inst.helloIdontExistSowhat); // valid no error because  Ctor2 is `any`
```

demanded:
```ts
 const Ctor2 = skate.define(class extends HTMLElement {
    static is = 'x-test-2';
    foo: number;
  });

const inst = new Ctor2();
const IamNumber: number = inst.foo; // valid, known property on Ctor2
console.log(inst.helloIdontExistSowhat); // compile error because  Ctor2 typeof is now properly inferred 
```

### PR DESCRIPTION
**Pull request by Hotell**
Closes #1050

### CODE REVIEW

File: src/ts-typings/api.d.ts [hunk 1]
```diff
@@ -67,10 +67,17 @@ export interface PropOptions<El, T> {
...
-};
+interface Define {
+  <T extends Partial<HTMLElement>>(ctor: T): T;
+  /**
+   *  @Deprecated - will be removed in 5.0
+   */
+  <T extends Partial<HTMLElement>>(name: string, ctor: T): T;
+}
+/**
+ * The define() function is syntactic sugar on top of customElements.define() that allows you to specify a static is property on your constructor that is the name of the component, or omit it altogether.
```
Comment (treshugart) [inline src/ts-typings/api.d.ts]: Do we need this comment? Seems superfluous as it's in the docs.

File: src/ts-typings/api.d.ts [hunk 2]
```diff
@@ -67,10 +67,17 @@ export interface PropOptions<El, T> {
...
+ */
+export var define: Define;
```
Comment (treshugart) [inline src/ts-typings/api.d.ts]: `const`?

File: src/ts-typings/api.d.ts (hunk 1 above)
Comment (Hotell) [inline src/ts-typings/api.d.ts]: the comment shows proper docs hint within IDE/Editor(vscode)

File: src/ts-typings/api.d.ts (hunk 2 above)
Comment (Hotell) [inline src/ts-typings/api.d.ts]: hmm, right. but I'm not 100% sure if in only `var` is supported in ambient definitions . Will take a look

Comment (treshugart) [inline src/ts-typings/api.d.ts]: That's kinda why I was asking. I'm not 100% either, but I would think it would be.

Comment (Hotell) [inline src/ts-typings/api.d.ts]: works with `const`

**Pull request merged**